api/src/test/java/org/pmiops/workbench/notebooks/NotebooksServiceImplTest.java

```diff
@@ -0,0 +1,98 @@
+package org.pmiops.workbench.notebooks;
+
+import static com.google.common.truth.Truth.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+import java.time.Clock;
+import org.json.JSONObject;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.pmiops.workbench.db.dao.UserRecentResourceService;
+import org.pmiops.workbench.db.model.User;
+import org.pmiops.workbench.firecloud.FireCloudService;
+import org.pmiops.workbench.firecloud.model.Workspace;
+import org.pmiops.workbench.firecloud.model.WorkspaceResponse;
+import org.pmiops.workbench.google.CloudStorageService;
+import org.pmiops.workbench.test.FakeClock;
+import org.pmiops.workbench.workspaces.WorkspaceService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+public class NotebooksServiceImplTest {
+
+  @TestConfiguration
+  @Import({NotebooksServiceImpl.class})
+  @MockBean({
+    CloudStorageService.class,
+    FireCloudService.class,
+    WorkspaceService.class,
+    UserRecentResourceService.class
+  })
+  static class Configuration {
+
+    @Bean
+    Clock clock() {
+      return new FakeClock();
+    }
+
+    @Bean
+    User user() {
+      return null;
+    }
+  }
+
+  @Autowired private NotebooksService notebooksService;
+  @Autowired private FireCloudService firecloudService;
+  @Autowired private CloudStorageService cloudStorageService;
+
+  @Test
+  public void testBasicContent() throws Exception {
```

Comment:
I would prepend all the tests with `readOnlyHtml_` to add some context. Otherwise looks good

Reply:
Oops, good catch - done.